include/vara/Feature/ConstraintParser.h

```diff
@@ -0,0 +1,353 @@
+    for (int Pos = 0; Pos < Cnt.size();) {
+      auto Result = munch(Pos);
+      TokenList.push_back(std::move(Result.first));
+      Pos += Result.second;
+      if (auto Kind = TokenList.back()->getKind();
+          Kind == ConstraintToken::ConstraintTokenKind::END_OF_FILE ||
+          Kind == ConstraintToken::ConstraintTokenKind::ERROR) {
+        break;
+      }
+    }
+
+    // ensure last token is always either EOF or an error
+    if (auto Kind = TokenList.back()->getKind();
+        Kind != ConstraintToken::ConstraintTokenKind::END_OF_FILE &&
+        Kind != ConstraintToken::ConstraintTokenKind::ERROR) {
+      TokenList.push_back(std::make_unique<ConstraintToken>(
+          ConstraintToken::ConstraintTokenKind::END_OF_FILE));
+    }
+
+    return std::move(TokenList);
+  }
+
+private:
+  using ResultTy = std::pair<std::unique_ptr<ConstraintToken>, int>;
+
+  std::string Cnt;
+
+  ResultTy munch(const int Pos) {
+    // TODO(se-passau/VaRA#664)
+    switch (Cnt[Pos]) {
+    case EOF:
+    case '\0':
+      return {std::move(std::make_unique<ConstraintToken>(
+                  ConstraintToken::ConstraintTokenKind::END_OF_FILE)),
+              1};
+    case ' ':
+    case '\t':
+    case '\n':
+    case '\r':
+      return munchWhitespace(Pos);
+    case '(':
+      return {std::move(std::make_unique<ConstraintToken>(
+                  ConstraintToken::ConstraintTokenKind::L_PAR)),
+              1};
+    case ')':
+      return {std::move(std::make_unique<ConstraintToken>(
+                  ConstraintToken::ConstraintTokenKind::R_PAR)),
+              1};
+    case '!':
+    case '+':
+    case '-':
+    case '<':
+    case '>':
+    case '~':
+    case '*':
+    case '&':
+    case '|':
+      return munchOperator(Pos);
+    case '0':
+    case '1':
+    case '2':
+    case '3':
+    case '4':
+    case '5':
+    case '6':
+    case '7':
+    case '8':
+    case '9':
+      return munchNumber(Pos);
+    case 'a':
+    case 'b':
+    case 'c':
+    case 'd':
+    case 'e':
+    case 'f':
+    case 'g':
+    case 'i':
+    case 'l':
+    case 'r':
+    case 's':
+    case 't':
+    case 'u':
+    case 'v':
+    case 'w':
+    case '_':
+    case 'h':
+    case 'j':
+    case 'k':
+    case 'm':
+    case 'n':
+    case 'o':
+    case 'p':
+    case 'q':
+    case 'x':
+    case 'y':
+    case 'z':
+    case 'A':
+    case 'B':
+    case 'C':
+    case 'D':
+    case 'E':
+    case 'F':
+    case 'G':
+    case 'H':
+    case 'I':
+    case 'J':
+    case 'K':
+    case 'L':
+    case 'M':
+    case 'N':
+    case 'O':
+    case 'P':
+    case 'Q':
+    case 'R':
+    case 'S':
+    case 'T':
+    case 'U':
+    case 'V':
+    case 'W':
+    case 'X':
+    case 'Y':
+    case 'Z':
+      return munchIdentifier(Pos);
+    default:
+      return {std::move(std::make_unique<ConstraintToken>(
+                  ConstraintToken::ConstraintTokenKind::ERROR,
+                  std::string(1, Cnt[Pos]))),
+              1};
```

Comment:
```suggestion
      return {std::make_unique<ConstraintToken>(
                  ConstraintToken::ConstraintTokenKind::ERROR,
                  std::string(1, Cnt[Pos])),
              1};
```
The created `unique_ptr` is already a rvalue so we don't need a move here.